### PR TITLE
[feat] Add many more downloadable dictionaries

### PR DESCRIPTION
**Pull request by Frenzie**
Virtually the entire list that was originally on Sourceforge, except for some dictionaries with unclear or questionable licences.

Many thanks to @avsej who prepared it all.

See https://github.com/koreader/koreader/pull/3176#issuecomment-447085441

### CODE REVIEW

File: frontend/ui/data/dictionaries.lua [hunk 1]
```diff
@@ -122,6 +122,1592 @@ local dictionaries = {
...
+    {
+        name = "Afrikaans-English dictionary",
+        lang_in = "Afrikaans",
+        lang_out = "English",
+        entries = 4198,
+        license = "GNU Public License",
+        url = "https://gitlab.com/avsej/dicts-stardict-form-xdxf/raw/264aadf8/002c/stardict-comn_sdict05_afrikaans-english-2.4.2.tar.bz2",
+    },
+    {
+        name = "Albanian-English dictionary",
```
Comment (Frenzie) [inline frontend/ui/data/dictionaries.lua]: We should probably change all of these to something like `FREELANG Lang A – Lang B dictionary` for clarity.

Comment (Frenzie) [reply frontend/ui/data/dictionaries.lua]: I was looking for the FREELANG license and it's not permissive. I'll send them an e-mail, but these may have to be excluded.

**Comment by Frenzie** [comment]
All good points, switching to plain tar probably makes the most sense since even best-case compression doesn't reduce it by more than about 5 %. (On the flipside, adding gzip compression here *also* reduces it by some 5 %, so an argument can certainly be made in its favor.)

As for `-a`, I believe that's only relevant when compressing? Whatever the case, it makes no difference for extracting on our GNU Tar 1.30. It automatically extracts the files correctly with `xf`.

**Comment by avsej** [comment]
the `.tar` files contain dictionaries in `.dz` format, which already gzipped. So no point in having `bzip2`

**Comment by Frenzie** [comment]
Something like `stardict-comn_sdict_axm05_Ukrainian_English-2.4.2` would be 6 MB without compression and is 2 MB with compression. The .dict.dz file may yield no or slightly negative returns, but the 4 MB .idx file compresses down to a little over 1 MB in gzip. And it goes down much further still with LZMA-type compressions:

screenshot_2018-12-14_11-46-52


So my 5 % was a slightly misguided example based on the specific dictionary I happened to be looking at, although it's likely to be generally more representative.

Now on my personal internet connection a few megabyte more or less is negligible, but the difference in filesize can certainly be quite pronounced.

**Comment by avsej** [comment]
I can repackage dicts in any format, but if koreader will guarantee that its `./tar` command will support it. How about bundling `xz` for example?

**Comment by Frenzie** [comment]
Thanks!

Note to self (and interested onlookers), link to FREELANG license for reference: https://www.freelang.net/dictionary/dic-copyrights.php

**Comment by poire-z** [comment]
You could add `keep_menu_open` (#4189) to the languages list menu, so we meet it back when leaving the KeyValuePage with dicts for one language.

**Comment by Frenzie** [comment]
Sure, I'll do that separately. :-)

**Comment by Frenzie** [comment]
Freelang reiterated their license. It seems that to "dissociate" dictionaries, the permission of individual authors has to be obtained.

**Pull request merged**